apps/angular/crud/src/app/item.component.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  inject,
+  input,
+} from '@angular/core';
+import { injectMutation } from '@tanstack/angular-query-experimental';
+import { lastValueFrom } from 'rxjs';
+import { Todo, TodosApiService } from './todos-api.service';
+
+@Component({
+  standalone: true,
+  selector: 'app-item',
+  template: `
+    <div>
+      @if (update.isError() || delete.isError()) {
+        <span>An error occur!</span>
+      } @else {
+        <span>{{ todo().title }}</span>
+      }
+
+      <button (click)="update.mutate(todo())">
+        @if (update.isPending()) {
+          <span>Updating...</span>
+        } @else {
+          <span>Update</span>
+        }
+      </button>
+      <button (click)="delete.mutate(todo().id)">
+        @if (delete.isPending()) {
+          <span>Deleting...</span>
+        } @else {
+          <span>Delete</span>
+        }
+      </button>
+    </div>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ItemComponent {
+  todo = input.required<Todo>();
+
+  #api = inject(TodosApiService);
+
+  update = injectMutation((client) => ({
+    mutationFn: (todo: Todo) => lastValueFrom(this.#api.updateTodo$(todo)),
+    onSuccess: () => {
+      client.invalidateQueries({ queryKey: ['todos'] });
+    },
+  }));
```

Comment:
I like to create another file called 'todo.query.ts' where I put all my related queries. it makes the code nicer.

Reply:
I also like to create a factory function for my keys.

Reply:
Yes good idea, factory function ? Why not an enum ?

Reply:
First I hate enums 😅, 
But then for your case, why not because todos is static, but then, you will need to add variables to create your key
```
queryKey: ['todos', todoId]
````